### ISSUE TITLE
conda install -c bioconda anndata2ri breaks import of scanpy

### ISSUE THREAD

**Issue by auesro** [scverse/scanpy#2172]
- [x] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the latest version of scanpy.
- [ ] (optional) I have confirmed this bug exists on the master branch of scanpy.

---
### Minimal code sample (that we can copy&paste without having any data)

```bash
conda create -n Scanpy python=3.7
conda activate Scanpy
conda install -c bioconda scanpy
conda install -c bioconda anndata2ri
```
```
import scanpy as sc
```

```pytb
Traceback (most recent call last):

  File "/tmp/ipykernel_2478/912249142.py", line 1, in <module>
    import scanpy as sc

  File "/home/augustoer/miniconda3/envs/Scanpy/lib/python3.7/site-packages/scanpy/__init__.py", line 15, in <module>
    from . import tools as tl

  File "/home/augustoer/miniconda3/envs/Scanpy/lib/python3.7/site-packages/scanpy/tools/__init__.py", line 17, in <module>
    from ._sim import sim

  File "/home/augustoer/miniconda3/envs/Scanpy/lib/python3.7/site-packages/scanpy/tools/_sim.py", line 23, in <module>
    from .. import _utils, readwrite, logging as logg

  File "/home/augustoer/miniconda3/envs/Scanpy/lib/python3.7/site-packages/scanpy/readwrite.py", line 10, in <module>
    import tables

  File "/home/augustoer/miniconda3/envs/Scanpy/lib/python3.7/site-packages/tables/__init__.py", line 62, in <module>
    from .file import File, open_file, copy_file

  File "/home/augustoer/miniconda3/envs/Scanpy/lib/python3.7/site-packages/tables/file.py", line 33, in <module>
    from . import hdf5extension

ImportError: /home/augustoer/miniconda3/envs/Scanpy/lib/python3.7/site-packages/tables/hdf5extension.cpython-37m-x86_64-linux-gnu.so: undefined symbol: H5Pset_fapl_direct
```
Removing anndata2ri with `conda remove anndata2ri` fixes the issue. Installing with `pip install anndata2ri` prevents the issue.

#### Versions
After fixing the issue.
<details>
WARNING: If you miss a compact list, please try `print_header`!
-----
anndata     0.7.8
scanpy      1.7.2
sinfo       0.3.1
-----
PIL                         9.0.1
PyQt5                       NA
anndata                     0.7.8
atomicwrites                1.4.0
autoreload                  NA
backcall                    0.2.0
beta_ufunc                  NA
binom_ufunc                 NA
bottleneck                  1.3.2
cffi                        1.15.0
chardet                     4.0.0
cloudpickle                 2.0.0
colorama                    0.4.4
cycler                      0.10.0
cython_runtime              NA
cytoolz                     0.11.0
dask                        2021.10.0
dateutil                    2.8.2
debugpy                     1.5.1
decorator                   5.1.1
defusedxml                  0.7.1
dunamai                     1.10.0
fsspec                      2022.01.0
get_version                 3.5.4
h5py                        2.10.0
igraph                      0.9.9
ipykernel                   6.4.1
ipython_genutils            0.2.0
jedi                        0.18.1
jinja2                      2.11.3
joblib                      1.1.0
kiwisolver                  1.3.2
legacy_api_wrap             0.0.0
leidenalg                   0.8.9
llvmlite                    0.37.0
markupsafe                  1.1.1
matplotlib                  3.5.1
matplotlib_inline           NA
mkl                         2.4.0
mpl_toolkits                NA
natsort                     7.1.1
nbinom_ufunc                NA
numba                       0.54.1
numexpr                     2.8.1
numpy                       1.20.3
packaging                   21.3
pandas                      1.3.4
parso                       0.8.3
pexpect                     4.8.0
pickleshare                 0.7.5
pkg_resources               NA
prompt_toolkit              3.0.20
psutil                      5.8.0
ptyprocess                  0.7.0
pydev_ipython               NA
pydevconsole                NA
pydevd                      2.6.0
pydevd_concurrency_analyser NA
pydevd_file_utils           NA
pydevd_plugins              NA
pydevd_tracing              NA
pygments                    2.11.2
pyparsing                   3.0.4
pytz                        2021.3
scanpy                      1.7.2
scipy                       1.7.3
setuptools_scm              NA
sinfo                       0.3.1
sip                         NA
six                         1.16.0
sklearn                     1.0.2
sphinxcontrib               NA
spyder                      5.1.5
spyder_kernels              2.1.3
spydercustomize             NA
storemagic                  NA
tables                      3.6.1
texttable                   1.6.4
threadpoolctl               2.2.0
tlz                         0.11.0
toolz                       0.11.2
tornado                     6.1
traitlets                   5.1.1
typing_extensions           NA
wcwidth                     0.2.5
wurlitzer                   3.0.2
yaml                        6.0
zipp                        NA
zmq                         22.3.0
-----
IPython             7.31.1
jupyter_client      6.1.12
jupyter_core        4.9.1
-----
Python 3.7.11 (default, Jul 27 2021, 14:32:16) [GCC 7.5.0]
Linux-5.4.0-104-generic-x86_64-with-debian-bullseye-sid
16 logical CPU cores, x86_64
-----
Session information updated at 2022-03-09 15:40

</details>

**Comment by ivirshup** [comment]
We currently distribute `scanpy` through `conda-forge`, so I would recommend using that for up to date versions.

Will this still error if you've done that?

Also, it's a little unclear if your session info came from the conda environment you're running into issues with. Is this definitely the case? There shouldn't be that new of releases of scanpy or anndata available through bioconda as far as I know.

**Comment by auesro** [comment]
I used `bioconda` as it was the recommended installation method according to the Scanpy documentation site last time I checked (october 2021). My bad then.

I only installed `scanpy`  through `bioconda` so the version I got (and you see in the Details of the previous post) are from there. `anndata2ri` version in there is from `pip`.

If I try with conda-forge channel for both scanpy and anndata2ri the issue is solved.
